public/components/register_model/model_task_manager.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Observable, timer, BehaviorSubject } from 'rxjs';
+import { takeWhile, switchMap } from 'rxjs/operators';
+import { APIProvider } from '../../apis/api_provider';
+import { TaskGetOneResponse } from '../../apis/task';
+
+interface TaskQueryOptions {
+  taskId: string;
+  onUpdate?: (status: TaskGetOneResponse) => void;
+  onError?: (err: Error) => void;
+  onComplete?: (status: TaskGetOneResponse) => void;
+}
+
+export class ModelTaskManager {
+  /**
+   * The model download tasks which are still running in BE
+   */
+  tasks = new BehaviorSubject<Map<string, Observable<TaskGetOneResponse>>>(new Map());
+
+  constructor() {}
+
+  remove(taskId: string) {
+    this.tasks.getValue().delete(taskId);
+    this.tasks.next(this.tasks.getValue());
+  }
+
+  query(options: TaskQueryOptions) {
+    if (!this.tasks.getValue().has(options.taskId)) {
+      const observable = timer(0, 2000)
+        .pipe(switchMap((_) => APIProvider.getAPI('task').getOne(options.taskId)))
+        // TODO: should it also check res.state?
+        // The intention here is to stop polling once a model is created
+        .pipe(takeWhile((res) => !Boolean(res.model_id), true));
```

Comment:
Do we need to change to code like these?
```ts
.pipe(takeWhile((res) => !Boolean(res.model_id) && !Boolean(res.error), true));
```
As I know, there `model_id` won't be exists if there is a model error response.